### PR TITLE
Add hdlib recipe

### PR DESCRIPTION
**Pull request by fabio-cumbo**
Hyperdimensional Computing Library for building Vector Symbolic Architectures in Python 3
[https://github.com/fabio-cumbo/hdlib](https://github.com/fabio-cumbo/hdlib)

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fabio-cumbo** [comment]
@conda-forge/staged-recipes PR is ready for review. Thanks!

File: recipes/hdlib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+test:
+  imports:
+    - hdlib
+
+about:
+  home: https://github.com/fabio-cumbo/hdlib
+  license: MIT
+  license_family: MIT
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (ocefpaf) [inline recipes/hdlib/meta.yaml:34]:
```suggestion
  license_file: LICENSE
```

Comment (ocefpaf) [reply recipes/hdlib/meta.yaml:34]: Latest conda-build can recognize the license in the recipe folder without that. If you saw that message in our CIs it is quite odd. If you saw that locally you need to upgrade your conda/conda-build.

Comment (fabio-cumbo) [reply recipes/hdlib/meta.yaml:34]: The error message appeared during CIs actually..

Comment (fabio-cumbo) [reply recipes/hdlib/meta.yaml]: It worked! Let me know if you think it could be merged now
Thanks @ocefpaf

**Pull request merged**